src/translations/mapCollection.js

```diff
@@ -0,0 +1,114 @@
+import { isUndefined, isDefined } from '../helpers/mixed';
+import { mixin } from '../helpers/object';
+import localHooks from '../mixins/localHooks';
+
+// Counter for checking if there is a memory leak.
+let registeredMaps = 0;
+
+class MapCollection {
+  constructor() {
+    this.mappings = new Map();
```

Comment:
I'd rather call it `this.baseMapCollection` to better explain the content of this map, and add a JSDoc that says what is the key and what is the value here.

Reply:
For discussion.

Reply:
For discussion.